gateway/xgress_edge/listener.go

```diff
@@ -296,6 +296,58 @@ func (proxy *ingressProxy) processUnbind(req *channel2.Message, ch channel2.Chan
 	}
 }
 
+func (proxy *ingressProxy) processUpdateBind(req *channel2.Message, ch channel2.Channel) {
+	token := string(req.Body)
+	log := pfxlog.ContextLogger(ch.Label()).WithField("sessionId", token).WithFields(edge.GetLoggerFields(req))
+
+	sm := fabric.GetStateManager()
+	ns := sm.GetNetworkSession(token)
+
+	if ns == nil {
+		log.WithField("token", token).Error("session not found")
+		proxy.sendStateClosedReply("Invalid Session", req)
+		return
+	}
+
+	if _, found := proxy.fingerprints.HasAny(ns.CertFingerprints); !found {
+		log.WithField("token", token).
+			WithField("serviceFingerprints", ns.CertFingerprints).
+			WithField("clientFingerprints", proxy.fingerprints.Prints()).
+			Error("matching fingerprint not found")
+		proxy.sendStateClosedReply("Invalid Session", req)
+		return
+	}
+
+	var cost *uint16
+	if costBytes, hasCost := req.Headers[edge.CostHeader]; hasCost {
+		updatedCost := binary.LittleEndian.Uint16(costBytes)
```

Comment:
should this just be a function ? `binary.LittleEndian` seems like the whole endian-ness stuff could be encapsulated. just a thought...

Reply:
We do have some utility methods on request that I'd forgotten about for put/get*IntHeader. We don't have one for uint16, but I'll do that in a follow up PR and do some tidying here.